lib/kramdown_rpf/kramdown.rb

```diff
@@ -6,202 +8,230 @@ module Converter
+        RPF::Plugin::Kramdown.convert_challenge_to_html(element.value)
       end
 
       # Convert :code_filename -> HTML
       # @api private
-      def convert_code(el, _indent)
-        RPF::Plugin::Kramdown.convert_code_to_html(el.value)
+      def convert_code(element, _indent)
+        RPF::Plugin::Kramdown.convert_code_to_html(element.value)
       end
 
       # Convert :collapse -> HTML
       # @api private
-      def convert_collapse(el, _indent)
-        RPF::Plugin::Kramdown.convert_collapse_to_html(el.value)
+      def convert_collapse(element, _indent)
+        RPF::Plugin::Kramdown.convert_collapse_to_html(element.value)
       end
 
       # Convert :hint -> HTML
       # @api private
-      def convert_hint(el, _indent)
-        RPF::Plugin::Kramdown.convert_hint_to_html(el.value)
+      def convert_hint(element, _indent)
+        RPF::Plugin::Kramdown.convert_hint_to_html(element.value)
       end
 
       # Convert :hints -> HTML
       # @api private
-      def convert_hints(el, _indent)
-        RPF::Plugin::Kramdown.convert_hints_to_html(el.value)
+      def convert_hints(element, _indent)
+        RPF::Plugin::Kramdown.convert_hints_to_html(element.value)
+      end
+
+      # Convert :knowledge_quiz_question -> HTML
+      # @api private
+      def convert_knowledge_quiz_question(element, indent)
+        RPF::Plugin::Kramdown.convert_knowledge_quiz_question_to_html(element.value, indent)
       end
 
       # Convert :new_page -> HTML
       # @api private
-      def convert_new_page(_el, _indent)
+      def convert_new_page(*)
         RPF::Plugin::Kramdown.convert_new_page_to_html
       end
 
       # Convert :no_print -> HTML
       # @api private
-      def convert_no_print(el, _indent)
-        RPF::Plugin::Kramdown.convert_no_print_to_html(el.value)
+      def convert_no_print(element, _indent)
+        RPF::Plugin::Kramdown.convert_no_print_to_html(element.value)
       end
 
       # Convert :print_only -> HTML
       # @api private
-      def convert_print_only(el, _indent)
-        RPF::Plugin::Kramdown.convert_print_only_to_html(el.value)
+      def convert_print_only(element, _indent)
+        RPF::Plugin::Kramdown.convert_print_only_to_html(element.value)
       end
 
       # Convert :quiz -> HTML
       # @api private
-      def convert_quiz(el, _indent)
-        RPF::Plugin::Kramdown.convert_quiz_to_html(el.value)
+      def convert_quiz(element, _indent)
+        RPF::Plugin::Kramdown.convert_quiz_to_html(element.value)
       end
 
       # Convert :save -> HTML
       # @api private
-      def convert_save(_el, _indent)
+      def convert_save(*)
         RPF::Plugin::Kramdown.convert_save_to_html
       end
 
       # Convert :task -> HTML
       # @api private
-      def convert_task(el, _indent)
-        RPF::Plugin::Kramdown.convert_task_to_html(el.value)
+      def convert_task(element, _indent)
+        RPF::Plugin::Kramdown.convert_task_to_html(element.value)
       end
     end
 
```

Comment:
I'm not clear why we have all these methods defined in the next two classes, but they could be shortened to:

```ruby
    # Stupid module name
    module RaiseNotImplementedIfNotImplemented
      def method_missing(method_name, *args, &block)
        raise NotImplementedError if method_name.to_s.start_with?('convert_')

        super
      end

      def respond_to_missing?(method_name, include_private = false)
        method_name.to_s.start_with?('convert_') || super
      end
    end

    class Kramdown
      include RaiseNotImplementedIfNotImplemented
    end
```

Reply:
Frankly I'd be happy to leave them as NoMethodError, but heh.